scripts/language_model/transformer/model.py

```diff
@@ -0,0 +1,90 @@
+# coding: utf-8
+
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+# pylint: disable=wildcard-import, arguments-differ
+"""Module for pre-defined NLP models."""
+
+import gluonnlp as nlp
+
+from .transformer import TransformerXL
+
+__all__ = ['get_model']
+
+
+def get_model(name, **kwargs):
+    """Returns a pre-defined model by name."""
+    models = {
+        # TODO better naming scheme when moving this to main API?
+        'transformerxl': transformerxl,
```

Comment:
maybe transformer_xl?

Reply:
The problem here is that different architectures are used under the same name "transformer_xl" for different datasets. We could postfix the dataset to the name (eg. `transformer_xl_wt103`, `transformer_xl_text8` etc.), but this would make the `dataset=` argument of our `get_model` function quite awkward.
Naming all the different models the same and returning different models depending on the dataset name is possible but I think we haven't done this so far. It may be unexpected.

Reply:
Are these architectures different due to different embedding dimensions, or something else?

Reply:
They vary across several parameters, such as number of layers, dimensions, which parameters are shared, how adaptive embedding / softmax is set up, ...: https://github.com/kimiyoung/transformer-xl/blob/master/tf/sota/enwik8.sh https://github.com/kimiyoung/transformer-xl/blob/master/tf/sota/lm1b.sh https://github.com/kimiyoung/transformer-xl/blob/master/tf/sota/text8.sh https://github.com/kimiyoung/transformer-xl/blob/master/tf/sota/wt103.sh